src/adapters/fiat-exchange-rates.ts

```diff
@@ -0,0 +1,53 @@
+import { fiatExchangeRatesErrorParser, fiatExchangeRatesParser } from "src/adapters/parsers";
+import { FIAT_EXCHANGE_RATES_API_URL } from "src/constants";
+import { Currency, FiatExchangeRates } from "src/domain";
+
+export interface GetFiatExchangeRatesResponse {
+  rates: Record<string, number>;
+}
+
+export interface GetFiatExchangeRatesError {
+  error: {
+    code: string;
+    message: string;
+  };
+}
+
+interface GetFiatExchangeRatesParams {
+  apiKey: string;
+}
+
+const getFiatExchangeRates = ({
+  apiKey,
+}: GetFiatExchangeRatesParams): Promise<FiatExchangeRates> => {
```

Comment:
What about that discussion we had regarding the return of typed errors `Promise<Either<FiatExchangeRates, GetFiatExchangeRatesError>>`?

Reply:
We agreed to give this a try on a different branch.